src/game_battlealgorithm.h

```diff
@@ -401,6 +406,7 @@ class AlgorithmBase {
 	bool killed_by_attack_damage;
 	bool critical_hit;
 	bool absorb;
+	bool revived = false;
```

Comment:
Because an attack can have multiple targets all state variables must be reset in the Reset() function.

Reply:
Great catch. This has been fixed.